task/buildah-oci-ta/0.4/buildah-oci-ta.yaml

```diff
@@ -313,37 +324,322 @@ spec:
           value: $(params.COMMIT_SHA)
         - name: DOCKERFILE
           value: $(params.DOCKERFILE)
+        - name: REUSE_COMPARISON_EXCLUSIONS
+          value: $(params.REUSE_COMPARISON_EXCLUSIONS)
       script: |
         #!/bin/bash
         set -euo pipefail
-
         echo "[$(date --utc -Ins)] Update CA trust"
-
         ca_bundle=/mnt/trusted-ca/ca-bundle.crt
         if [ -f "$ca_bundle" ]; then
           echo "INFO: Using mounted CA bundle: $ca_bundle"
-          cp -vf $ca_bundle /etc/pki/ca-trust/source/anchors
+          cp -vf "$ca_bundle" /etc/pki/ca-trust/source/anchors
           update-ca-trust
         fi
-
+        echo "[$(date --utc -Ins)] Calculate tree hash and check for reusable artifacts"
+        # Calculate tree hash from git commit
+        if [ -n "$COMMIT_SHA" ]; then
+          # Run git from the source directory without changing working directory
+          TREE_HASH=$(git -C "/var/workdir/source" show -s --format="%T" "$COMMIT_SHA")
+          echo "Tree hash for commit $COMMIT_SHA: $TREE_HASH"
+          # Store tree hash for reuse in other steps
+          echo "$TREE_HASH" >"/var/workdir/tree-hash"
+          # Emit the tree hash result
+          echo -n "$TREE_HASH" | tee "$(results.GIT_TREE_HASH.path)"
+        else
+          echo "No COMMIT_SHA provided, skipping artifact reuse"
+          TREE_HASH=""
+          # Store empty tree hash for reuse in other steps
+          echo "" >"/var/workdir/tree-hash"
+          # Emit empty tree hash result
+          echo -n "" | tee "$(results.GIT_TREE_HASH.path)"
+        fi
+        # Check for reusable artifacts if we have a tree hash
+        if [ -n "$TREE_HASH" ]; then
+          echo "Checking for reusable artifacts with tree hash: $TREE_HASH"
+          # Parse the exclusion parameters from the YAML array
+          EXCLUSION_PARAMS=()
+          while IFS= read -r line; do
+            if [[ "$line" =~ ^[[:space:]]*-[[:space:]]*([A-Z_]+)$ ]]; then
+              EXCLUSION_PARAMS+=("${BASH_REMATCH[1]}")
+            fi
+          done <<<"$REUSE_COMPARISON_EXCLUSIONS"
+          echo "Parameters to exclude from comparison: ${EXCLUSION_PARAMS[*]}"
+          # Check if tree hash tag exists for this architecture
+          TREE_TAG="tree-$TREE_HASH"
+          IMAGE_REPO="${IMAGE%:*}"
+          CURRENT_ARCH=$(uname -m)
+          echo "Checking for tree tag: $TREE_TAG on $IMAGE_REPO for architecture: $CURRENT_ARCH"
+          # Find all candidates with the tree hash tag
+          echo "Searching for all candidates with tree hash tag: $TREE_TAG"
+          # Get all tags and filter for our tree hash tag
+          ALL_TAGS=$(skopeo list-tags "docker://$IMAGE_REPO" | jq -r '.Tags[]' 2>/dev/null || echo "")
```

Comment:
with big repos listing all tags will take forever, if you know tag in advance why don't just check if tag exist directly?

Reply:
I agree, I have reviewed this and removed all the 'candidate selection' logic. not needed